src/compiler/checker.ts

```diff
@@ -2530,7 +2530,7 @@ namespace ts {
                     : error(name, Diagnostics.Module_0_declares_1_locally_but_it_is_not_exported, moduleName, declarationName);
 
                 addRelatedInfo(diagnostic,
-                    createDiagnosticForNode(localSymbol.valueDeclaration, Diagnostics._0_is_declared_here, declarationName)
+                    createDiagnosticForNode(localSymbol.valueDeclaration ?? localSymbol.declarations[0], Diagnostics._0_is_declared_here, declarationName)

```

Comment:
There could be more than one declaration, but exporting _any_ of them will fix the error, so 🤷‍♂

Reply:
You could also report an error on all of them which might feel cleaner and work well too.

Reply:
I'd probably just avoid the `...map` and use `forEach` but whatever you'd prefer.